python-sdk/src/astro/sql/operators/dataframe.py

```diff
@@ -148,17 +148,21 @@ def execute(self, context: Context) -> Table | pd.DataFrame:
             self.op_kwargs, self.python_callable, self.columns_names_capitalization
         )
 
-        pandas_dataframe = self.python_callable(*self.op_args, **self.op_kwargs)
-        pandas_dataframe = convert_columns_names_capitalization(
-            df=pandas_dataframe,
-            columns_names_capitalization=self.columns_names_capitalization,
-        )
+        function_output = self._convert_column_capitlization_for_output()
         if self.output_table:
+            self.log.debug(
+                "Output table found, converting function output to SQL table"
+            )
+            if not isinstance(function_output, pd.DataFrame):
+                raise ValueError(
+                    "Astro can only turn a single dataframe into a table. Please change your"
+                    "function output."
+                )
```

Comment:
Can we add a test case for this code block

Reply:
added